caffe2/operators/batch_gather_ops.h

```diff
@@ -4,6 +4,8 @@
 #include "caffe2/core/context.h"
 #include "caffe2/core/operator.h"
 #include "caffe2/utils/math.h"
+// Reuse helper logic from GatherOp since BatchGather is the same with axis=1.
+#include "caffe2/operators/gather_op.h"
 
 namespace caffe2 {
 
```

Comment:
so the only difference between the functions right now is the default value for `axis`, you could template the entire class on `int default_axis` and register it twice

Reply:
Hmm, you mean get rid of batch_gather_op.h/cpp and reuse the class for both registrations? I've considered merging the sources... 

Actually, is there a way to dynamically check within the gather and gradient op what registration they come from, so as to reuse the code without a template?